### PR TITLE
Adds switch-layer rendering functionality for f.hair, glasses and head accessories.

### PR DESCRIPTION
**Pull request by KasparoVy**
Glasses will now overlay hair only in certain circumstances. **This was mainly written to address an issue where, previously, we couldn't have glasses not be covered by Vulpkanin markings and remain under hair at the same time**. In addition to that fix, glasses will now be rendered ontop of all sufficiently short hairstyles.

_From left to right: Glasses covering short hair, glasses being covered by long hair, ditto, glasses behaving with Vulpkanin markings, glasses arms covered by Unathi facial frills._
select-layer rendering and glasses in action

Hair that is short (has glasses_over set) will have glasses rendered on top of it rather than the default behaviour of below it.

This happens to tie in with a feature PR I've got iced, so it killed a few birds with one stone.

:cl:
rscadd: You can now define whether a hair style will cover glasses or not. Only long hair styles should cover glasses (default behaviour).
rscadd: You can now define whether a facial hair style/head accessory will be rendered on top of hair or not.
fix: Glasses no longer render on top of all hair styles, only sufficiently short ones.
/:cl:

### CODE REVIEW

**Comment by FlattestGuitar** [comment]
You could go as far as to say that this fixes  #5839

**Comment by KasparoVy** [comment]
Wow, that's nifty! Apparently a recent update to GitHub included some sort of automatic conflict management feature. I just straight up merged from master and there were no conflicts, whereas if I did so a while back I'd say with certainty there would be. Nice!

File: code/__DEFINES/misc.dm [hunk 1]
```diff
@@ -211,21 +211,24 @@
...
+#define FACEMASK_LAYER			25
+#define HEAD_LAYER				26
+#define COLLAR_LAYER			27
+#define HANDCUFF_LAYER			28
+#define LEGCUFF_LAYER			29
+#define L_HAND_LAYER			30
+#define R_HAND_LAYER			31
+#define TARGETED_LAYER			32	//BS12: Layer for the target overlay from weapon targeting system
+#define FIRE_LAYER				33	//If you're on fire
+#define TOTAL_LAYERS			33
```
Comment (Crazylemon64) [inline code/__DEFINES/misc.dm]: 3 layers added:
`HEAD_ACC_OVER_LAYER`
`FHAIR_OVER_LAYER`
`GLASSES_OVER_LAYER`
(just making notes)

**Comment by Spacemanspark** [comment]
Please merge this, IPCs look terrible with glasses on right now.

**Comment by KasparoVy** [comment]
Just going to shamelessly copypaste myself here

This won't 'fix' IPCs looking ugly with glasses. The reason why this is is because the last PR that went through (that totally refactored markings yet again) fixed a bug where, previously, you'd only see a few pixels on either side of an IPC's screen due to hair always covering glasses (among many other things, although it also created some other issues that get corrected in this PR).

The only remedy for this would be to make sprites for IPCs that turn glasses into 'modules' that clip onto the sides of their heads at eye-level. It'll look similar to the way it did before, but make a whole lot more sense and look a fair bit better. I'll leave that for a soon(tm) PR as I'm dealing with final exams RN and spriting is some of my slowest work. Furthermore, it'll require me to implement some login ontop of the species-fitting and sprite-sheets-ing in the update_inv_glasses proc in update_icons.dm that checks whether the player's head is a monitor or not if the player's head is robotic.

**Pull request merged**